fe/fe-core/src/main/java/org/apache/doris/datasource/es/QueryBuilders.java

```diff
@@ -266,7 +267,21 @@ public static QueryBuilder toEsDsl(Expr expr, List<Expr> notPushDownList, Map<St
         // Replace col with col.keyword if mapping exist.
         column = fieldsContext.getOrDefault(column, column);
         if (expr instanceof BinaryPredicate) {
-            return parseBinaryPredicate(expr, opCode, column, needDateCompat);
+            BinaryPredicate binaryPredicate = (BinaryPredicate) expr;
+            Expr value;
+            if (binaryPredicate.slotIsLeft()) {
+                value = binaryPredicate.getChild(1);
+            } else {
+                value = binaryPredicate.getChild(0);
+            }
+            // only push down literal expr to ES
+            if (value instanceof LiteralExpr) {
+                LiteralExpr literalExpr = (LiteralExpr) value;
+                return parseBinaryPredicate(literalExpr, opCode, column, needDateCompat);
```

Comment:
if binary predicate is literal vs slot, u should flip opCode

Reply:
done